lib/components/app/locale-selector.tsx

```diff
@@ -44,31 +51,42 @@ const LocaleSelector = (props: LocaleSelectorProps): JSX.Element => {
   }
 
   return (
-    <NavDropdown
+    <Dropdown
       id="locale-selector"
-      pullRight
-      title={
-        <StyledIconWrapper style={{ color: 'rgba(255, 255, 255, 0.85)' }}>
-          <GlobeAmericas />
-        </StyledIconWrapper>
+      label={intl.formatMessage({ id: 'components.SubNav.selectALanguage' })}
+      listLabel={intl.formatMessage({ id: 'components.SubNav.languages' })}
+      name={
+        <span
+          style={{
+            color: 'rgba(255, 255, 255, 0.85)'
+          }}
+        >
+          <GlobeAmericas height="18px" />
+        </span>
       }
+      // TODO: How to make this work without block ruby?
+      style={{ display: 'block ruby' }}
     >
       {Object.keys(configLanguages)
         .filter((locale) => locale !== 'allLanguages')
         .map((locale) => (
-          <MenuItem
-            className="locale-name"
+          <li
+            aria-selected={locale === currentLocale}
             key={locale}
             onClick={(e: MouseEvent) => handleLocaleSelection(e, locale)}
+            onKeyPress={(e: any) => handleLocaleSelection(e, locale)}
```

Comment:
Make these handlers point to a single callback (`useCallback`).

Reply:
Whoops, almost forgot this one, add the `lang` attribute to this element!

Reply:
What would this callback do? Both the locale and event are different every time